src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,38 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+	public String value;
+	private boolean isPrivate;
+	private String messageConstraints;
+	private String validationRegex;
+	
+	public Contact(String value, boolean isPrivate
+			, String messageConstraints, String validationRegex)
+					throws IllegalValueException {
+		this.isPrivate = isPrivate;
+		this.messageConstraints = messageConstraints;
+		this.validationRegex = validationRegex;
+		if (!isValidInfo(value.trim())){
+			throw new IllegalValueException(this.messageConstraints);
+		}
+		this.value = value.trim();
+	}
+	
+	public boolean isValidInfo(String test){
+		return test.matches(validationRegex);
+	}
+	
+	public String toString() {
```

Comment:
I guess you should mark these methods are overridden methods.
From [Java documentation](https://docs.oracle.com/javase/tutorial/java/annotations/predefined.html):

> While it is not required to use this annotation when overriding a method, it helps to prevent errors. If a method marked with @Override fails to correctly override a method in one of its superclasses, the compiler generates an error.